pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/worker/WorkerAssignmentStrategyFactory.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner.logical.worker;
+
+import java.util.Map;
+import org.apache.pinot.query.routing.WorkerManager;
+
+
+public class WorkerAssignmentStrategyFactory {
+  private final WorkerManager _workerManager;
+
+  public WorkerAssignmentStrategyFactory(WorkerManager workerManager) {
+    _workerManager = workerManager;
+  }
+
+  public WorkerAssignmentStrategy create(Map<String, String> queryOptions, long requestId) {
+    return new DefaultWorkerAssignmentStrategy(_workerManager, requestId);
+  }
```

Comment:
I made two improvements in the latest revision:

1. Made `WorkerAssignmentStrategyFactory::create` static and made the class static as well by making the ctor private. This is similar to `QuerySchedulerFactory`. I feel it's neat this way since it doesn't make sense to have to pass around both the WorkerAssignmentFactory and WorkerManager across classes. UT may become an issue later but we can deal with it once it becomes an issue.
2. I have kept `WorkerManager` as a required dependency for creating a WorkerAssignmentStrategy and the `requestId` is passed to the method call `assignWorkers` instead of being a member var of the strategy.